### PR TITLE
Mobius function

### PR DESCRIPTION
**Pull request by pradyotprakash**
Mobius function with some basic tests added.
@isuruf Can you review this?

### CODE REVIEW

File: symengine/ntheory.cpp [hunk 1]
```diff
@@ -1474,4 +1474,29 @@ void powermod_list(std::vector<RCP<const Integer>> &pows, const RCP<const Intege
     }
 }
 
+int mobius(const Integer &a)
```
Comment (Sumith1896) [inline symengine/ntheory.cpp]: Isn't it better if this is a SymEngine Integer?

File: symengine/tests/ntheory/test_ntheory.cpp [hunk 2]
```diff
@@ -635,3 +635,28 @@ TEST_CASE("test_powermod(): ntheory", "[ntheory]")
...
+    REQUIRE(mobius(*i2) == -1);
+    REQUIRE(mobius(*i3) == -1);
+    REQUIRE(mobius(*i4) == 0);
+    REQUIRE(mobius(*i5) == -1);
+    REQUIRE(mobius(*i6) == 1);
+    REQUIRE(mobius(*i7) == -1);
+    REQUIRE(mobius(*i8) == 0);
+    REQUIRE(mobius(*i9) == 0);
+    REQUIRE(mobius(*i10) == 1);
+}
```
Comment (Sumith1896) [inline symengine/tests/ntheory/test_ntheory.cpp]: new line

**Comment by Sumith1896** [comment]
Looks good to me.

File: symengine/ntheory.cpp (hunk 1 above)
Comment (Sumith1896) [inline symengine/ntheory.cpp]: @isuruf What do you think?

Comment (isuruf) [inline symengine/ntheory.cpp]: @Sumith1896, what do you mean?

Comment (Sumith1896) [inline symengine/ntheory.cpp]: Return type as a `Integer` instead of machine int. Would be better in actual usage.

Comment (isuruf) [inline symengine/ntheory.cpp]: Not really necessary as the return value is always 1, 0, or -1

Comment (Sumith1896) [inline symengine/ntheory.cpp]: Cool

**Comment by certik** [comment]
Can you please add a documentation as a comment into the header file about what the definition of the Mobius function? I assume it's the one from here: https://en.wikipedia.org/wiki/M%C3%B6bius_function

**Comment by pradyotprakash** [comment]
@certik Does this seem fine now?

**Comment by certik** [comment]
That looks good. Do you want to squash the commits into just one?

**Comment by pradyotprakash** [comment]
@certik I merged the commits that I could. I can't do the ones above merge with master. Is this fine or I can send in a PR with just one commit.

**Comment by abhinavagarwal07** [comment]
@pradyotprakash 
you can use `git rebase -i HEAD~x` to squash

**Pull request closed without merging**